build.gradle

```diff
@@ -60,3 +61,42 @@ jacocoTestReport {
 }
 
 build.dependsOn jacocoTestReport
+
+pmd {
+    toolVersion = '5.4.2'
+    ignoreFailures = true
+    sourceSets = [sourceSets.main]
+    ruleSets = []
+    ruleSetFiles = files('pmd.xml')
+}
+
+task('pmd', type: Pmd, dependsOn: 'assemble') {
```

Comment:
This task is not necessary. Do you want a task to run manually? If so, just run `./gradlew pmdMain`.